node/src/components/linear_chain.rs

```diff
@@ -438,52 +439,49 @@ where
                     return Effects::new();
                 }
                 self.add_pending_finality_signature(*fs.clone());
-                match self.block_cache.get(&block_hash) {
-                    None => effect_builder.get_block_from_storage(block_hash).event(
-                        move |maybe_block| {
-                            let maybe_box_block = maybe_block.map(Box::new);
-                            Event::GetBlockForFinalitySignaturesResult(fs, maybe_box_block)
-                        },
-                    ),
-                    Some(block) => effect_builder.immediately().event(move |_| {
-                        Event::GetBlockForFinalitySignaturesResult(fs, Some(Box::new(block)))
+                match self.signature_cache.get(&block_hash) {
+                    None => effect_builder
+                        .get_signatures_from_storage(block_hash)
+                        .event(move |maybe_signatures| {
+                            let maybe_box_signatures = maybe_signatures.map(Box::new);
+                            Event::GetStoredFinalitySignaturesResult(fs, maybe_box_signatures)
+                        }),
+                    Some(signatures) => effect_builder.immediately().event(move |_| {
+                        Event::GetStoredFinalitySignaturesResult(fs, Some(Box::new(signatures)))
                     }),
                 }
             }
-            Event::GetBlockForFinalitySignaturesResult(fs, maybe_block) => {
-                if let Some(block) = &maybe_block {
-                    if block.header().era_id() != fs.era_id {
+            Event::GetStoredFinalitySignaturesResult(fs, maybe_signatures) => {
+                if let Some(signatures) = &maybe_signatures {
+                    if signatures.era_id != fs.era_id {
                         warn!(public_key=%fs.public_key, "Finality signature with invalid era id.");
                         // TODO: Disconnect from the sender.
                         return Effects::new();
                     }
-                    // Populate cache so that next finality signatures don't have to read from the
-                    // storage. If block is already from cache then this will be a noop.
-                    self.block_cache.insert(*block.clone())
+                    // Populate cache
```

Comment:
Please bring back the full comment and update if necessary.

Reply:
Brought back full comment